src/PrestaShopBundle/EventListener/TokenizedUrlsListener.php

```diff
@@ -101,9 +101,14 @@ public function onKernelRequest(GetResponseEvent $event)
             }
         }
 
-        $token = urldecode($request->query->get('_token', false));
+        $token = false;
+        if ($request->query->has('_token')) {
+            $token = new CsrfToken($this->username, $request->query->get('_token'));
+        } elseif ($request->query->has('form')) {
+            $token = new CsrfToken('form', $request->query->get('form')['_token']);
```

Comment:
Are you sure there's always a `_token` inside `form`?

Reply:
yes because of this https://github.com/PrestaShop/PrestaShop/blob/develop/app/config/config.yml#L30 but I'll add a new check to be sure the key exists